spec/05_select_random_elements_from_array_spec.rb

```diff
@@ -0,0 +1,30 @@
+# frozen_string_literal: true
+
+# Specification for RubyMonk Primer Problem 5 Select random elements from an array
+# Note that the tests implemented here are not the same as Ruby Monk's
+
+require_relative '../05_select_random_elements_from_array'
+
+RSpec.describe SelectRandomElementsFromArray do
+  subject(:array_elements) { described_class.new }
+
+  describe '#random_select' do
+    it 'passing [1, 2, 3, 4, 5] and 2 returned' do
+      expect(array_elements.random_select([1, 2, 3, 4, 5], 2).count).to be 2
+    end
+
+    it 'returns only the values present in the array passed to the method' do
+      og_array = [1, 2, 3, 4, 5]
+      results = array_elements.random_select(og_array, 3)
+      expect(Set.new(results).subset?(Set.new(og_array))).to be true
```

Comment:
Why did you get `Set` involved here?  I don't think it's needed.

Here are my top two preferred methods of checking whether a subset is in an array:
- `(og_array & results) == results` is extremely idiomatic, so it might not be preferable here
- `results.each { |i| expect(og_array).to include i }` is more verbose, but also more explicit

It's weird to me that `Set` implements `#subset` but `Enumerable` does not.  SMH.

As a side note, the description here is not helpful to me.  Are you trying to say that it will only select values that are in the given array?

Reply:
> results.each { |i| expect(og_array).to include i }

Another option is
```ruby
expect(og_array).to contain_exactly(results)
```

Reply:
Oh nevermind, I misunderstood the test. I'd probably do

```ruby
expect(og_array).to include(*results)
```

Reply:
So I used `Set` because this to me was inherently a set and subset relationship (although I agree, not the most clearly written description which is directly from RubyMonk's test).  Duplicates and order (array characteristics) are irrelevant for the intent of this test.  My brain also often thinks in terms of set theory so I may be biased.

Reply:
Minor note, `contain_exactly` ignores ordering, and `include(*results)` will also ignore ordering while also not caring about duplicates. I use the latter a lot at work, though I should use the former more

Reply:
Oh, I didn't know about `#contain_exactly`!

I also didn't realize that the `include` matcher allowed multiple values.  Slick.  TIL.  His way is better than mine; I quite like `include(*results)` here.

Re: `Set` specifically -- you are right about set/subset, and that's probably why arrays don't have a `#subset?` method (good call!).  However, in my experience `Set` is only used in place of an `Array` when it specifically matters that the list is unordered.  YMMV.  I'm curious if @Serneum agrees?

If this were feature code, there might be other benefits of working with `Sets` directly, but in this particular test it only serves to add a new library, and superfluous data transformation -- additional overhead to anyone trying to use the tests as documentation.

Reply:
I don't think I've ever used `Set` in 6 years of doing Ruby dev, to be honest

Reply:
This is great, I am really learning some stuff here.  So I did notice the `#contain_exactly` matcher and didn't think that applied to this case and per this test it doesn't (which [Serneum](https://github.com/Serneum) noted above), still I thought it'd be fun to test and see the results.

So here's the SET (structured Exploratory Test) that I added to this file
```
    it 'testing contain_exactly matcher with dups' do
      original_array_with_duplicate = [1, 2, 3, 4, 5, 2]
      results_with_duplicates = [2, 2, 2]
      expect(original_array_with_duplicate).to contain_exactly(results_with_duplicates)
    end
```
And here's the failure...
```
Failures:

  1) SelectRandomElementsFromArray#random_select testing contain_exactly matcher with dups
     Failure/Error: expect(original_array_with_duplicate).to contain_exactly(results_with_duplicates)

       expected collection contained:  [[2, 2, 2]]
       actual collection contained:    [1, 2, 2, 3, 4, 5]
       the missing elements were:      [[2, 2, 2]]
       the extra elements were:        [1, 2, 2, 3, 4, 5]
     # ./spec/05_select_random_elements_from_array_spec.rb:33:in `block (3 levels) in <top (required)>'
```

And (not so fun fact), the RSpec test runner extension (Ruby Test Runner) that I added to my VS Code to run tests in the IDE does not recognize the `specify` keyword 🤦

Reply:
>I don't think I've ever used `Set` in 6 years of doing Ruby dev, to be honest

LOL, yeah.  I have an affinity for set theory and have seen it used often when dealing with arrays when ordering and uniqueness is not a consideration.  It also provides basic math and logic operators such as `+`, `-`, and `&` which can be helpful.  Basically if you have an array where order does not matter and duplicates are not allowed, strongly consider using a `Set` instead as that is basically what it is and not an array.  That said, your reality may prove differently.

Reply:
FWIW, Arrays also define `+`, `-`, and `&`.

Personally, for almost all of the arrays I create, order does not matter and duplicates get removed via `uniq` if it matters.  I never saw the benefits of working with `Sets` since `Array` can do everything, but you may yet convince me!

It would make more sense to me if `Set` were the default and we only moved to an `Array` if order *did* matter, or we *did* need to allow duplicates, but that's not what ruby (or perhaps, the ruby community) has decided on.

Reply:
> Arrays also define +, -, and &.

Came here to say that as well. I don't recall the last time order mattered in my arrays outside of returning data, and usually I do that at the database level

Reply:
Nerd Fight!  🤣 

I think that if your shared context and past experience and practice (aka using arrays with unique values and where order does not matter) is well understood, then probably adding in the use of `Set` just adds cognitive load and belies the value.  For me, I learned `Set` while learning Ruby so I prefer the absolute clarity in knowing that it does contain unique values and order doesn't matter.  I don't need to look at any other code to know this, just the use of `Set` alone.   TL;DR: Y'all are super nerds and y'all should do y'all.

Reply:
Nerd fights are the best fights. 🥊